docs/walkthrough/redshift-estimator.rst

```diff
@@ -0,0 +1,470 @@
+
+.. figure:: images/redshift-inputs.png
+    :align: center
+    :scale: 50%
+
+Each **Input** operation requires an artifact to have been added to the project. To do this, go to the artifacts view and click either of the two floating buttons in the bottom right of the workspace (one button only appears on hover).
+
+.. figure:: images/artifact-blank.png
+    :align: center
+    :scale: 50%
+
+With these buttons, you can either upload a local file to one of the storage backends or import a file that already exists within a storage backend.
+
+.. figure:: images/artifact-import-upload.png
+    :align: center
+    :scale: 50%
+
+By default, all artifacts are treated as python `pickle objects <https://docs.python.org/3/library/pickle.html>`_. Using other forms of serialized data, such as `FITS <https://fits.gsfc.nasa.gov/fits_documentation.html>`_ or `npy <https://numpy.org/doc/stable/reference/generated/numpy.lib.format.html>`_ files, requires defining a custom serializer in the *Custom Serialization* view, which is not covered in this tutorial.
+
+TrainRedshift Operation
+=======================
+
+The first custom operation will create and train the neural network classifier.
+
+Two attributes should be added: *batch_size* and *epochs*. Batch size is the number of training samples that the model will be trained on at a time and epochs is the number of times that each training sample will be given to the model. Both are important hyperparameters for a neural network. For this guide, the attributes are defined as shown below, but the exact number used for default values can be changed as desired by the reader.
+
+This operation will require two inputs (images and labels) and a neural network architecture. Finally, the operation produces one output, which is the trained classifier model. After all inputs, outputs, and attributes have been added, the structure of the operation should appear similar to the following:
+
+.. figure:: images/redshift-train-io.png
+    :align: center
+    :scale: 50%
+
+The code for this operation follows the standard procedure for creating and training a Keras network with one minor caveat. Keras models expect the output of a network during training to be an array or an array of indices. Because the outputs used in this tutorial are floating point values, they must be converted into one of these forms. This is the purpose of the *to_categorical* function below. The code for this operation is shown below.
```

Comment:
> Keras models expect the output of a network during training to be an array or an array of indices. Because the outputs used in this tutorial are floating point values, they must be converted into one of these forms. This is the purpose of the to_categorical function below. The code for this operation is shown below.

My understanding was the biggest motivation for this function was due to changing this from a regression problem to a classification problem (ie, ensuring the output of the network was a probability distribution).

Reply:
I believe I buried the lede on this section. That is the motivator for the function, but I was attempting to describe why the original data is not immediately compatible with classification training. I agree that the original wording is confusing and distracting. I have changed it to the below

> The code for this operation follows the standard procedure for creating and training a Keras network with one minor caveat. The method used by Pasquet et al. on which this pipeline is based formulates redshift prediction as a classification problem. Because the labels used in this tutorial are floating point values, they must be converted into a categorical format. This is the purpose of the *to_categorical* function. The code for this operation is shown below.